numpy/polynomial/chebyshev.py

```diff
@@ -1830,6 +1831,73 @@ def chebcompanion(c):
     return mat
 
 
+def chebinterp(func, deg, x=None, args=(), domain=None, window=None):
+    """Approximate a function using Chebyshev polynomial interpolation.
+
+    Return the Chebyshev series instance that interpolates a given function.
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the fitting Chebyshev polynomials. If `deg` is a single
+        integer all terms up to and including the `deg`'th term are included in
+        the interpolation.
+    x : array_like, shape (M,), optional
+        Defines window of interpolation. The window of the series
+        is given by the lowest and highest members of this list.
+    args : tuple, optional
+        Extra arguments to be used in the function call.
+    domain : [beg, end], optional
+        Unused. Overridden to `[-1, 1]` for Chebyshev polynomial interpolation.
+    window : [beg, end], optional
+        Window to use for the returned series. The default value is the
+        extrema of `x`.
+
+    Returns
+    -------
+    new_series : series
+        A series that represents the polynomial interpolation
+        to the function.
+
+
+    Examples
+    --------
+    >>> from numpy import tanh, pi
+    >>> import numpy.polynomial.chebyshev as C
+    >>> C.chebinterp(lambda x: tanh(x)+0.5, [0, 2*pi], 8)
+    array([ 1.30198814,  0.3540514 , -0.25209078,  0.14032589, -0.05706644,
+            0.01226391,  0.00367748, -0.00455335])
+
+    .. versionadded:: 1.14.0
+
+    """
+    warnings.warn("Defaulting to Chebyshev nodes and domain for "
```

Comment:
If domain can only be [-1,1] then it isn't obvious that this should be an input.  If it would be something else (e.g. a strict subinterval [0,1]) then I think the correct behavior would be the raise ValueError on an invalid domain since most functions don't use magic to override user requests.